project/dexter-monitor/public/ctrl/projectManagementCtrl.js

```diff
@@ -25,11 +25,109 @@
  */
 "use strict";
 
-monitorApp.controller("ProjectManagementCtrl", function($scope) {
+monitorApp.controller("ProjectManagementCtrl", function($scope, $mdDialog, $log, $location, $sce, ProjectService) {
 
     initialize();
 
     function initialize() {
     }
+		$scope.$watch('newProject', function(newProject){
+			$scope.status =  $sce.trustAsHtml('Server will be created at <a href="' + newProject.hostName + ":" + newProject.portNumber + "/#/\">" + newProject.hostName + ":" + newProject.portNumber + "</a>");
+		},true);
+	
+		$scope.languages = ["CPP", "JAVA"];
+		$scope.newProject = {hostName: $location.host(), portNumber:""};
+					
+		$scope.showCreateProjectDialog = function(ev) {
+						
+			$mdDialog.show({
+			controller: "ProjectManagementCtrl",
+			  templateUrl: '/view/createServerDialog.html',
+			  parent: angular.element(document.querySelector('#popupContainer')),
+			  targetEvent: ev,
+			  clickOutsideToClose:true
+			});
+		};
+		
+		$scope.createServer = function(newProject) {
+			
+			validateProjectData(newProject).then(
+			(errorMessage) => {
```

Comment:
It would be better changed as below. 
You can reduce the depth of the this function.
```
if(errorMessage){   
  $log.error(~); 
  return; 
} ;
// do something:(not else scope) 
$scope.status = ''; 
```